src/ThemedWrappers.h

```diff
@@ -17,22 +17,28 @@
 #include <optional>
 #include <array>
 #include <map>
+#include <type_traits>
 
 #include "widgets/AButton.h"
 #include "Internat.h"
 #include "Observer.h"
 #include "Theme.h"
 #include "Prefs.h"
 
+struct PrefsListenerDummy { };
+
 template<class WindowBase>
 class ThemedWindowWrapper final :
    public WindowBase,
-   public PrefsListener
+   public std::conditional_t<std::is_base_of_v<PrefsListener, WindowBase>, PrefsListenerDummy, PrefsListener>
```

Comment:
Maybe this is more elegant:

```
template<class WindowBase, class... OtherBases>
class ThemedWindowWrapper final :
   public WindowBase, public OtherBases...,
```

So you can simply use `PrefsListener` where needed as another template argument, and not mention that class name here.